crates/router/src/connector/square/transformers.rs

```diff
@@ -312,15 +312,7 @@ impl TryFrom<&types::ConnectorAuthType> for SquareAuthType {
                 api_key: api_key.to_owned(),
                 key1: key1.to_owned(),
             }),
-
-            types::ConnectorAuthType::HeaderKey { .. }
-            | types::ConnectorAuthType::SignatureKey { .. }
-            | types::ConnectorAuthType::MultiAuthKey { .. }
-            | types::ConnectorAuthType::CurrencyAuthKey { .. }
-            | types::ConnectorAuthType::TemporaryAuth { .. }
-            | types::ConnectorAuthType::NoKey { .. } => {
-                Err(errors::ConnectorError::FailedToObtainAuthType.into())
-            }
+            _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
```

Comment:
We try to avoid defaults please do not make this change

Reply:
I don't think matching rest of the ConnectorAuthType is necessary in case of this enum. Any additional enum variant added to this enum will not affect this logic.